File: leetcode/0676-implement-magic-dictionary/MagicDictionary_test.go
package leetcode

import (
	"fmt"
	"path/filepath"
	"testing"

	"github.com/edram/leetcode/util"
)

type MagicDictionaryCase struct {
	Dictionary []string
	Cases      []struct {
		Input  string
		Output bool
	}
}

func TestMagicDictionary(t *testing.T) {
	var casePath, _ = filepath.Abs("./cases.json")
	payload, err := util.LoadCases[[]MagicDictionaryCase](casePath)
	if err != nil {
		panic(err)
	}

	for i, testCase := range *payload {
		fmt.Printf("正在执行第 %d 个测试用例\n", i+1)

		dictionary := Constructor()
		dictionary.BuildDict(testCase.Dictionary)

		for _, check := range testCase.Cases {
			result := dictionary.Search(check.Input)
			if check.Output != result {
				t.Errorf("期望 %v, 结果 %v", check.Output, result)
			}
		}

	}
}

File: leetcode/0676-implement-magic-dictionary/MagicDictionary.test.ts
import cases from "./cases.json";
import MagicDictionary from "./MagicDictionary";

test("implement magic dictionary", () => {
  cases.forEach((testCase, index) => {
    console.log(`正在执行第 ${index + 1} 个测试用例\n`);
    console.time("time");
    const dictionary = new MagicDictionary();
    dictionary.buildDict(testCase.dictionary);
    for (const item of testCase.cases) {
      expect(item.output).toEqual(dictionary.search(item.input));
    }
    console.timeEnd("time");
  });
});
